<a href="https://colab.research.google.com/github/jeremyjyang/unmchp/blob/main/colab/CPT_compare.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# CPT code list comparison

Author: Jeremy Yang

Comparison of CPT code lists from  
 - NM HCA (Dec 20, 2024 LOD)

and

 - CMS
   - 2023-09-29
   - 2024-10-18
   - 2025-07-09


In [78]:
import pandas as pd
from google.colab import sheets

In [79]:
from google.colab import data_table
data_table.enable_dataframe_formatter()

## Read file A

HCA LoD #42 (Dec 20, 2024) Attachment A, Table 1

In [80]:
url = 'https://raw.githubusercontent.com/jeremyjyang/unmchp/refs/heads/main/data/Final-LOD-42-Final-Biomarker-Coverage_TABLE1-Biomarker-Testing-Codes.tsv'
try:
  dfA = pd.read_csv(url, sep='\t')
  display(dfA.head())
except Exception as e:
  print(f"An error occurred while reading the file from URL: {e}")

Procedure Code                   Description
0          0004M            Scoliosis dna alys
1          0006M  Onc hep gene risk classifier
2          0007M   Onc gastro 51 gene nomogram
3          0011M  Onc prst8 ca mrna 12 gen alg
4          0012M   Onc mrna 5 gen rsk urthl ca

In [81]:
print(f"Rows: {dfA.shape[0]}; Columns: {dfA.shape[1]}")
dfA.columns = ["Code", "Description"]
unique_codes_countA = dfA['Code'].nunique()
print(f"The number of unique values in the 'Code' column is: {unique_codes_countA}")

Rows: 611; Columns: 2
The number of unique values in the 'Code' column is: 589


## Find duplicate codes

In [82]:
duplicate_rowsA = dfA[dfA.duplicated(subset=['Code'], keep=False)]
if not duplicate_rowsA.empty:
  print("Rows with duplicate 'Code' values:")
  display(duplicate_rowsA.sort_values(by='Code').head())
else:
  print("No duplicate values found in 'Code' column.")

Rows with duplicate 'Code' values:


Code                   Description
130  0215U  Rare ds xom dna alys ea comp
139  0215U  Rare ds xom dna alys ea comp
131  0216U   Neuro inh ataxia dna 12 com
140  0216U   Neuro inh ataxia dna 12 com
141  0217U  Neuro inh ataxia dna 51 gene

## Read file B

Billing and Coding: Molecular Pathology and Genetic Testing (A58917) (July 9, 2025)

https://www.cms.gov/medicare-coverage-database/view/article.aspx?articleId=58917&ver=83

In [83]:
url = 'https://raw.githubusercontent.com/jeremyjyang/unmchp/refs/heads/main/data/Article_-_Billing_and_Coding_Molecular_Pathology_and_Genetic_Testing_A58917_20250709_CODING-INFORMATION.tsv'
try:
    dfB = pd.read_csv(url, sep='\t')
    display(dfB.head())
except Exception as e:
    print(f"An error occurred while reading the file from URL: {e}")

Code       Description
0  81105  Hpa-1 genotyping
1  81106  Hpa-2 genotyping
2  81107  Hpa-3 genotyping
3  81108  Hpa-4 genotyping
4  81109  Hpa-5 genotyping

In [84]:
print(f"Rows: {dfB.shape[0]}; Columns: {dfB.shape[1]}")
unique_codes_countB = dfB['Code'].nunique()
print(f"The number of unique values in the 'Code' column is: {unique_codes_countB}")

Rows: 553; Columns: 2
The number of unique values in the 'Code' column is: 553


## Find duplicate codes

Set B

In [85]:
duplicate_rowsB = dfB[dfB.duplicated(subset=['Code'], keep=False)]
if not duplicate_rowsB.empty:
  print("Rows with duplicate 'Code' values:")
  display(duplicate_rowsB.sort_values(by='Code'))
else:
  print("No duplicate values found in 'Code' column.")

No duplicate values found in 'Code' column.


## Compare code sets

A versus B

In [86]:
setA = set(dfA['Code'])
setB = set(dfB['Code'])

intersectionAB = setA.intersection(setB)
differenceAB = setA.difference(setB)
differenceBA = setB.difference(setA)

In [87]:
print(f"Number of unique codes in dfA: {len(setA)}")
print(f"Number of unique codes in dfB: {len(setB)}")
print(f"Number of codes in intersection: {len(intersectionAB)}")
print(f"Number of codes in dfA but not in dfB: {len(differenceAB)}")
print(f"Number of codes in dfB but not in dfA: {len(differenceBA)}")

Number of unique codes in dfA: 589
Number of unique codes in dfB: 553
Number of codes in intersection: 535
Number of codes in dfA but not in dfB: 54
Number of codes in dfB but not in dfA: 18


In [88]:
print(f"Intersection of codes ({len(intersectionAB)}): {', '.join(intersectionAB)}")

Intersection of codes (535): 81233, 81267, 81173, 81341, 0341U, 81300, 0419U, 81170, 81292, 0073U, 81349, 0047U, 0177U, 81220, 81273, 81509, 81350, 0291U, 81189, 0270U, 0266U, 0363U, 81321, 0076U, 0229U, 0326U, 81343, 0193U, 0268U, 81311, 81255, 81219, 81302, 0410U, 81284, 81425, 0027U, 0197U, 81426, 81313, 81416, 81204, 0288U, 0470U, 0140U, 0060U, 81404, 81320, 81194, 0398U, 0084U, 81352, 0191U, 0362U, 81265, 0130U, 81332, 0252U, 0403U, 81183, 81406, 81405, 0055U, 81448, 81362, 81596, 0049U, 81315, 0089U, 81266, 0006M, 81254, 81431, 81110, 81276, 81309, 81190, 0334U, 81234, 0420U, 81297, 0156U, 81210, 0411U, 81108, 81293, 81525, 81528, 0152U, 81400, 0131U, 81346, 0250U, 81107, 81163, 0184U, 0187U, 0068U, 0001U, 81318, 81539, 0265U, 0260U, 81335, 0007U, 81288, 81456, 0195U, 0379U, 0371U, 81236, 81268, 0258U, 0332U, 81164, 0134U, 0046U, 0317U, 0211U, 0242U, 0343U, 81345, 81275, 81294, 81412, 81317, 81225, 81202, 0096U, 0433U, 0335U, 0333U, 0339U, 81121, 81245, 81325, 81375, 0094U, 0413U

In [89]:
print(f"Codes in dfA but not in dfB ({len(differenceAB)}): {', '.join(differenceAB)}")

Codes in dfA but not in dfB (54): 0486U, 0467U, 88269, 0345U, 88267, G9143, 0078U, 0462U, 0469U, 0369U, 88248, 0460U, 0465U, 0459U, 88262, 88275, 0488U, 88372, 88280, 81438, 81560, 88245, 88261, 0461U, 0471U, 88283, 88371, 0020M, 0374U, 88272, 0474U, 0464U, 88274, 88285, 81433, 88249, 88264, 0373U, 81436, 0489U, 0380U, 0456U, 88263, 0475U, 0487U, 88273, 0472U, 88289, 0458U, 0457U, 0370U, 88271, 0396U, 0468U


In [90]:
print(f"Codes in dfB but not in dfA ({len(differenceBA)}): {', '.join(differenceBA)}")

Codes in dfB but not in dfA (18): 81463, 81195, 0494U, 81462, 0485U, 81558, 0560U, 0481U, 81441, 81459, 81457, 81464, 81458, 0561U, 0246U, 0562U, 0478U, 81449


In [91]:
dfB[dfB['Code'].isin(differenceBA)].head()

Code                   Description
43   81195  Cytog genom-wid alys hem mal
245  81441   Ibmfs seq alys pnl 30 genes
250  81449     So neo gsap 5-50 rna alys
255  81457    So neo gsap dna mcrstl ins
256  81458   So gsap dna cpy nmbr&mcrstl

## Read file C

Billing and Coding: Molecular Pathology and Genetic Testing (A58917) (October 18, 2024)

https://www.cms.gov/medicare-coverage-database/view/article.aspx?articleid=58917&ver=77&

In [92]:
url = 'https://raw.githubusercontent.com/jeremyjyang/unmchp/refs/heads/main/data/Article_-_Billing_and_Coding_Molecular_Pathology_and_Genetic_Testing_A58917_20241018_CODING-INFORMATION.tsv'
try:
    dfC = pd.read_csv(url, sep='\t')
    display(dfC.head())
except Exception as e:
    print(f"An error occurred while reading the file from URL: {e}")

Code       Description
0  81105  Hpa-1 genotyping
1  81106  Hpa-2 genotyping
2  81107  Hpa-3 genotyping
3  81108  Hpa-4 genotyping
4  81109  Hpa-5 genotyping

In [93]:
print(f"Rows: {dfC.shape[0]}; Columns: {dfC.shape[1]}")
unique_codes_countC = dfC['Code'].nunique()
print(f"The number of unique values in the 'Code' column is: {unique_codes_countC}")

Rows: 556; Columns: 2
The number of unique values in the 'Code' column is: 556


## Find duplicate codes

Set C

In [94]:
duplicate_rowsC = dfC[dfC.duplicated(subset=['Code'], keep=False)]
if not duplicate_rowsC.empty:
  print("Rows with duplicate 'Code' values:")
  display(duplicate_rowsC.sort_values(by='Code'))
else:
  print("No duplicate values found in 'Code' column.")

No duplicate values found in 'Code' column.


## Compare code sets

A versus C

In [95]:
setC = set(dfC['Code'])
intersectionAC = setA.intersection(setC)
differenceAC = setA.difference(setC)
differenceCA = setC.difference(setA)

In [96]:
print(f"Number of unique codes in dfA: {len(setA)}")
print(f"Number of unique codes in dfC: {len(setC)}")
print(f"Number of codes in intersection: {len(intersectionAC)}")
print(f"Number of codes in dfA but not in dfC: {len(differenceAC)}")
print(f"Number of codes in dfC but not in dfA: {len(differenceCA)}")

Number of unique codes in dfA: 589
Number of unique codes in dfC: 556
Number of codes in intersection: 543
Number of codes in dfA but not in dfC: 46
Number of codes in dfC but not in dfA: 13


In [97]:
print(f"Intersection of codes ({len(intersectionAC)}): {', '.join(intersectionAC)}")

Intersection of codes (543): 81233, 81267, 81173, 81341, 0341U, 81300, 0419U, 81170, 81292, 0073U, 81349, 0047U, 0177U, 81220, 81273, 81509, 81350, 0291U, 81189, 0270U, 0266U, 0363U, 81321, 0076U, 0229U, 0326U, 81343, 0193U, 0268U, 81311, 81255, 81219, 81302, 0410U, 81284, 81425, 0027U, 0197U, 81426, 81313, 81416, 81204, 0288U, 0470U, 0140U, 0060U, 81404, 81320, 81194, 0398U, 0084U, 81352, 0191U, 0362U, 81265, 0130U, 81332, 0252U, 0403U, 81183, 81406, 81405, 0055U, 81448, 81362, 81596, 0049U, 81315, 0089U, 81266, 0006M, 81254, 81431, 81110, 81276, 81309, 81190, 0334U, 81234, 0420U, 81297, 0156U, 81210, 0411U, 81108, 81293, 81525, 81528, 0152U, 81400, 0131U, 81346, 0250U, 0374U, 81107, 81163, 0184U, 0187U, 0068U, 0001U, 81318, 81539, 0265U, 0260U, 0370U, 81335, 0007U, 81288, 81456, 0195U, 0379U, 0371U, 81236, 81268, 0258U, 0332U, 81164, 0134U, 0046U, 0317U, 0211U, 0242U, 0343U, 81345, 81275, 81294, 81412, 81317, 81225, 81202, 0096U, 0433U, 0335U, 0333U, 0339U, 81121, 81245, 81325, 81375

In [98]:
print(f"Codes in dfA but not in dfC ({len(differenceAC)}): {', '.join(differenceAC)}")

Codes in dfA but not in dfC (46): 0486U, 0467U, 88269, 0345U, 88267, G9143, 0078U, 0462U, 0469U, 88248, 0460U, 0465U, 0459U, 88262, 88275, 0488U, 88372, 88280, 81560, 88245, 88261, 0461U, 0471U, 88283, 88371, 0020M, 88272, 0474U, 0464U, 88274, 88285, 88249, 88264, 0489U, 0456U, 88263, 0475U, 0487U, 88273, 0472U, 88289, 0458U, 0457U, 88271, 0396U, 0468U


In [99]:
print(f"Codes in dfC but not in dfA ({len(differenceCA)}): {', '.join(differenceCA)}")

Codes in dfC but not in dfA (13): 81463, 0494U, 81462, 0485U, 81459, 0481U, 81441, 81457, 81464, 81458, 0246U, 0478U, 81449


In [100]:
dfC[dfC['Code'].isin(differenceCA)].head(10)

Code                   Description
247  81441   Ibmfs seq alys pnl 30 genes
252  81449     So neo gsap 5-50 rna alys
257  81457    So neo gsap dna mcrstl ins
258  81458   So gsap dna cpy nmbr&mcrstl
259  81459       So neo gsap dna/dna&rna
261  81462    So gsap cll fr dna/dna&rna
262  81463  So gsap cl fr cpy nmbr&mcrst
263  81464     So gsap cll fr mcrstl ins
456  0246U  Rbc dna gnotyp 16 bld groups
552  0478U  Onc nsclc dna&rna dpcr 9gens

## Read file D

Billing and Coding: Molecular Pathology and Genetic Testing (A58917) (September 29, 2023)

https://localcoverage.cms.gov/mcd_archive/view/article.aspx?articleInfo=58917:62&

In [101]:
url = 'https://raw.githubusercontent.com/jeremyjyang/unmchp/refs/heads/main/data/Article_-_Billing_and_Coding_Molecular_Pathology_and_Genetic_Testing_A58917_20230929_CODING-INFORMATION.tsv'
try:
    dfD = pd.read_csv(url, sep='\t')
    display(dfD.head())
except Exception as e:
    print(f"An error occurred while reading the file from URL: {e}")

Code       Description  Unnamed: 2
0  81105  Hpa-1 genotyping         NaN
1  81106  Hpa-2 genotyping         NaN
2  81107  Hpa-3 genotyping         NaN
3  81108  Hpa-4 genotyping         NaN
4  81109  Hpa-5 genotyping         NaN

In [102]:
print(f"Rows: {dfD.shape[0]}; Columns: {dfD.shape[1]}")
unique_codes_countD = dfD['Code'].nunique()
print(f"The number of unique values in the 'Code' column is: {unique_codes_countD}")

Rows: 540; Columns: 3
The number of unique values in the 'Code' column is: 540


## Find duplicate codes

Set D

In [103]:
duplicate_rowsD = dfD[dfD.duplicated(subset=['Code'], keep=False)]
if not duplicate_rowsD.empty:
  print("Rows with duplicate 'Code' values:")
  display(duplicate_rowsD.sort_values(by='Code'))
else:
  print("No duplicate values found in 'Code' column.")

No duplicate values found in 'Code' column.


## Compare code sets

A versus D

In [104]:
setD = set(dfD['Code'])
intersectionAD = setA.intersection(setD)
differenceAD = setA.difference(setD)
differenceDA = setD.difference(setA)

In [105]:
print(f"Number of unique codes in dfA: {len(setA)}")
print(f"Number of unique codes in dfD: {len(setD)}")
print(f"Number of codes in intersection: {len(intersectionAD)}")
print(f"Number of codes in dfA but not in dfD: {len(differenceAD)}")
print(f"Number of codes in dfD but not in dfA: {len(differenceDA)}")

Number of unique codes in dfA: 589
Number of unique codes in dfD: 540
Number of codes in intersection: 536
Number of codes in dfA but not in dfD: 53
Number of codes in dfD but not in dfA: 4


In [106]:
print(f"Intersection of codes ({len(intersectionAD)}): {', '.join(intersectionAD)}")

Intersection of codes (536): 81233, 81267, 81173, 81341, 0341U, 81300, 0419U, 81170, 81292, 0073U, 81349, 0047U, 0177U, 81220, 81273, 81509, 81350, 0291U, 81189, 0270U, 0266U, 0363U, 81321, 0076U, 0229U, 0326U, 81343, 0193U, 0268U, 81311, 81255, 81219, 81302, 0410U, 81284, 81425, 0027U, 0197U, 81426, 81313, 81416, 81204, 0288U, 0140U, 0060U, 81404, 81320, 81194, 0398U, 0084U, 81352, 0191U, 0362U, 81265, 0130U, 81332, 0252U, 0403U, 81183, 81406, 81405, 0055U, 81448, 81362, 81596, 0049U, 81315, 0089U, 81266, 0006M, 81254, 81431, 81110, 81276, 81309, 0334U, 81190, 81234, 81297, 0156U, 81210, 0411U, 81108, 81293, 81525, 81528, 0152U, 81400, 0131U, 81346, 0250U, 0374U, 81107, 81163, 0184U, 0187U, 0068U, 0001U, 81318, 81539, 0265U, 0260U, 0370U, 81335, 0007U, 81288, 81456, 0195U, 0379U, 0371U, 81236, 81268, 0258U, 0332U, 81164, 0134U, 0046U, 0317U, 0211U, 0242U, 0343U, 81345, 81275, 81294, 81412, 81317, 81225, 81202, 0096U, 0335U, 0333U, 0339U, 81121, 81245, 81325, 81375, 0094U, 0413U, 81529

In [107]:
print(f"Codes in dfA but not in dfD ({len(differenceAD)}): {', '.join(differenceAD)}")

Codes in dfA but not in dfD (53): 0420U, 0486U, 0466U, 0467U, 0463U, 88269, 0345U, 88267, G9143, 0424U, 0470U, 0462U, 0469U, 88248, 0460U, 0465U, 0459U, 88262, 88275, 0488U, 88372, 88280, 81560, 88245, 88261, 0461U, 0471U, 88283, 0020M, 88371, 88272, 0474U, 0464U, 88274, 88285, 88249, 88264, 0444U, 0423U, 0489U, 0456U, 88263, 0475U, 0487U, 88273, 0472U, 88289, 0473U, 0458U, 0457U, 0433U, 88271, 0468U


In [108]:
print(f"Codes in dfD but not in dfA ({len(differenceDA)}): {', '.join(differenceDA)}")

Codes in dfD but not in dfA (4): 0204U, 0246U, 81441, 81449


In [109]:
dfD[dfD['Code'].isin(differenceDA)]

Code                   Description  Unnamed: 2
247  81441   Ibmfs seq alys pnl 30 genes         NaN
252  81449    Tgsap so neo 5-50 rna alys         NaN
423  0204U  Onc thyr mrna xprsn alys 593         NaN
452  0246U  Rbc dna gnotyp 16 bld groups         NaN

## Dereferencing CPT codes

Codes can be dereferenced for full descriptions, via UMLS, with an API KEY, using links such as:

 - [81441](https://cts.nlm.nih.gov/fhir/res/CodeSystem/$lookup?system=http%3A%2F%2Fwww.ama-assn.org%2Fgo%2Fcpt&version=2025&_format=json&code=81441), [81449](https://cts.nlm.nih.gov/fhir/res/CodeSystem/$lookup?system=http%3A%2F%2Fwww.ama-assn.org%2Fgo%2Fcpt&version=2025&_format=json&code=81449), [0204U](https://cts.nlm.nih.gov/fhir/res/CodeSystem/$lookup?system=http%3A%2F%2Fwww.ama-assn.org%2Fgo%2Fcpt&version=2025&_format=json&code=0204U), [0246U](https://cts.nlm.nih.gov/fhir/res/CodeSystem/$lookup?system=http%3A%2F%2Fwww.ama-assn.org%2Fgo%2Fcpt&version=2025&_format=json&code=0246U), [0031U](https://cts.nlm.nih.gov/fhir/res/CodeSystem/$lookup?system=http%3A%2F%2Fwww.ama-assn.org%2Fgo%2Fcpt&version=2025&_format=json&code=0031U)

In [110]:
df_diffDA = dfD[dfD['Code'].isin(differenceDA)]
df_diffDA = df_diffDA[['Code', 'Description']]
df_diffDA['UMLS_Link'] = df_diffDA['Code'].str.replace("^", "https://cts.nlm.nih.gov/fhir/res/CodeSystem/$lookup?system=http%3A%2F%2Fwww.ama-assn.org%2Fgo%2Fcpt&version=2025&_format=json&code=", regex=True)
df_diffDA.head()

Code                   Description  \
247  81441   Ibmfs seq alys pnl 30 genes   
252  81449    Tgsap so neo 5-50 rna alys   
423  0204U  Onc thyr mrna xprsn alys 593   
452  0246U  Rbc dna gnotyp 16 bld groups   

                                             UMLS_Link  
247  https://cts.nlm.nih.gov/fhir/res/CodeSystem/$l...  
252  https://cts.nlm.nih.gov/fhir/res/CodeSystem/$l...  
423  https://cts.nlm.nih.gov/fhir/res/CodeSystem/$l...  
452  https://cts.nlm.nih.gov/fhir/res/CodeSystem/$l...

In [111]:
df_diffAD = dfA[dfA['Code'].isin(differenceAD)]
df_diffAD = df_diffAD[['Code', 'Description']]
df_diffAD['UMLS_Link'] = df_diffAD['Code'].str.replace("^", "https://cts.nlm.nih.gov/fhir/res/CodeSystem/$lookup?system=http%3A%2F%2Fwww.ama-assn.org%2Fgo%2Fcpt&version=2025&_format=json&code=", regex=True)
df_diffAD.head()

Code                   Description  \
8    0020M   Onc cns alys 30000 dna loci   
219  0345U    PSYC GENOM ALYS PNL 15 GEN   
262  0420U    Onc urthl mrna xprsn 6 snp   
263  0423U  Psyc genomic alys pnl 26 gen   
264  0424U  Onc prst8 xom alys 53 sncrna   

                                             UMLS_Link  
8    https://cts.nlm.nih.gov/fhir/res/CodeSystem/$l...  
219  https://cts.nlm.nih.gov/fhir/res/CodeSystem/$l...  
262  https://cts.nlm.nih.gov/fhir/res/CodeSystem/$l...  
263  https://cts.nlm.nih.gov/fhir/res/CodeSystem/$l...  
264  https://cts.nlm.nih.gov/fhir/res/CodeSystem/$l...

## Read UMLS file of CPT codes

In [112]:
url = 'https://raw.githubusercontent.com/jeremyjyang/unmchp/refs/heads/main/data/Biomarker-Testing-Codes_bioclients-umls-api-out-selectedcols.tsv'
try:
    df_umls = pd.read_csv(url, sep='\t')
    display(df_umls.head())
except Exception as e:
    print(f"An error occurred while reading the file from URL: {e}")

CPT_id rootSource                                               name  \
0  0004M        CPT  Scoliosis, DNA analysis of 53 single nucleotid...   
1  0006M        CPT  Oncology (hepatic), mRNA expression levels of ...   
2  0007M        CPT  Oncology (gastrointestinal neuroendocrine tumo...   
3  0011M        CPT  Oncology, prostate cancer, mRNA expression ass...   
4  0012M        CPT  Oncology (urothelial), mRNA, gene expression p...   

  obsolete  atomCount           concepts  
0    False        4.0  C3869478;C5697507  
1    False        4.0  C3869477;C5697508  
2    False        4.0  C3869476;C5697509  
3    False        4.0  C4732724;C5697510  
4    False        4.0  C5697511;C4732725

In [113]:
df_umls = df_umls[['CPT_id', 'name']]
df_umls.columns = ['Code', 'Name']
df_umls.head()

Code                                               Name
0  0004M  Scoliosis, DNA analysis of 53 single nucleotid...
1  0006M  Oncology (hepatic), mRNA expression levels of ...
2  0007M  Oncology (gastrointestinal neuroendocrine tumo...
3  0011M  Oncology, prostate cancer, mRNA expression ass...
4  0012M  Oncology (urothelial), mRNA, gene expression p...

In [114]:
df_diffDA = pd.merge(df_diffDA, df_umls, on='Code', how='left')
df_diffDA[['Code', 'Description', 'Name']].head(100)

Code                   Description Name
0  81441   Ibmfs seq alys pnl 30 genes  NaN
1  81449    Tgsap so neo 5-50 rna alys  NaN
2  0204U  Onc thyr mrna xprsn alys 593  NaN
3  0246U  Rbc dna gnotyp 16 bld groups  NaN

In [115]:
df_diffAD = pd.merge(df_diffAD, df_umls, on='Code', how='left')
df_diffAD[['Code', 'Description', 'Name']].head(100)

Code                   Description  \
0   0020M   Onc cns alys 30000 dna loci   
1   0345U    PSYC GENOM ALYS PNL 15 GEN   
2   0420U    Onc urthl mrna xprsn 6 snp   
3   0423U  Psyc genomic alys pnl 26 gen   
4   0424U  Onc prst8 xom alys 53 sncrna   
5   0433U   Onc prst8 5 dna reg mrk pcr   
6   0444U    Onc sld orgn neo tgsap 361   
7   0456U   Ai ra ngs 19 genes anti-ccp   
8   0457U   Pfas 9 cmpnd lcms/ms pls/sr   
9   0458U  Onc brst ca s100 a8&a9 elisa   
10  0459U      Abeta42 & ttau eclia csf   
11  0460U  Onc whl bld/bucc rtpcr 24gen   
12  0461U  Onc rxgenom alys rtpcr 24gen   
13  0462U  Melatonin lvl tst slp std7/9   
14  0463U  Onc crvx mrna genxprsn 14bmk   
15  0464U   Onc clrct scr qrtsa dna mrk   
16  0465U  Onc urthl carc dna qmsp 2gen   
17  0466U   Crd cad dna gwas 564856 snp   
18  0467U  Onc bldr dna ngs 60gen&aneup   
19  0468U   Hep nash mir34a5p a2m ykl40   
20  0469U  Rare ds whl gen seq ftl samp   
21  0470U  Onc orop detcj mrd 8 dna hpv   
22  0471U  Onc clrc ca 35 vrn kras&nras   
23  0472U    Ca vi psp&sp1 antb sjogren   
24  0473U  Onc sld tum bld/slv 648 gene   
25  0474U  Hered pan ca gsap 88gene ngs   
26  0475U  Hered prst8 ca gsap 23 genes   
27  0486U   ONC PAN SOL TUM NGS CFCTDNA   
28  0487U   ONC SOL TUM CFCDNA TGSAP 84   
29  0488U   OB FETAL AG NIPT CFDNA ALYS   
30  0489U   OB SGNIPT CFDNA SEQ ALYS 1+   
31  81560  TRNSPLJ PD LVR&BWL CD154+CLL   
32  88245     CHROMOSOME ANALYSIS 20-25   
33  88248    CHROMOSOME ANALYSIS 50-100   
34  88249       CHROMOSOME ANALYSIS 100   
35  88261         CHROMOSOME ANALYSIS 5   
36  88262     CHROMOSOME ANALYSIS 15-20   
37  88263        CHROMOSOME ANALYSIS 45   
38  88264     CHROMOSOME ANALYSIS 20-25   
39  88267    CHROMOSOME ANALYS PLACENTA   
40  88269    CHROMOSOME ANALYS AMNIOTIC   
41  88271        CYTOGENETICS DNA PROBE   
42  88272              CYTOGENETICS 3-5   
43  88273            CYTOGENETICS 10-30   
44  88274            CYTOGENETICS 25-99   
45  88275          CYTOGENETICS 100-300   
46  88280    CHROMOSOME KARYOTYPE STUDY   
47  88283      CHROMOSOME BANDING STUDY   
48  88285   CHROMOSOME COUNT ADDITIONAL   
49  88289   CHROMOSOME STUDY ADDITIONAL   
50  88371   PROTEIN WESTERN BLOT TISSUE   
51  88372      PROTEIN ANALYSIS W/PROBE   
52  G9143  WARFARIN RESPON GENETIC TEST   

                                                 Name  
0   Oncology (central nervous system), analysis of...  
1   Psychiatry (eg, depression, anxiety, attention...  
2   Oncology (urothelial), mRNA expression profili...  
3   Psychiatry (eg, depression, anxiety), genomic ...  
4   Oncology (prostate), exosome-based analysis of...  
5   Oncology (prostate), 5 DNA regulatory markers ...  
6   Oncology (solid organ neoplasia), targeted gen...  
7   Autoimmune (rheumatoid arthritis), next-genera...  
8   Perfluoroalkyl substances (PFAS) (eg, perfluor...  
9   Oncology (breast cancer), S100A8 and S100A9, b...  
10  β-amyloid (Abeta42) and total tau (tTau), elec...  
11  Oncology, whole blood or buccal, DNA single-nu...  
12  Oncology, pharmacogenomic analysis of single-n...  
13  Melatonin levels test, sleep study, 7 or 9 sam...  
14  Oncology (cervix), mRNA gene expression profil...  
15  Oncology (colorectal) screening, quantitative ...  
16  Oncology (urothelial carcinoma), DNA, quantita...  
17  Cardiology (coronary artery disease [CAD]), DN...  
18  Oncology (bladder), DNA, next-generation seque...  
19  Hepatology (nonalcoholic steatohepatitis [NASH...  
20  Rare diseases (constitutional/heritable disord...  
21  Oncology (oropharyngeal), detection of minimal...  
22  Oncology (colorectal cancer), qualitative real...  
23  Carbonic anhydrase VI (CA VI), parotid specifi...  
24  Oncology (solid tumor), next-generation sequen...  
25  Hereditary pan-cancer (eg, hereditary sarcomas...  
26  Hereditary prostate cancer-related disorders, ...  
27  Oncology (pan-solid tumor), next-generation se...  
28  Oncology (solid tumor), cell-free circulating ...  
29